pkg/controller/argocd/service.go

```diff
@@ -393,7 +394,17 @@ func (r *ReconcileArgoCD) reconcileServerMetricsService(cr *argoprojv1a1.ArgoCD)
 // reconcileServerService will ensure that the Service is present for the Argo CD server component.
 func (r *ReconcileArgoCD) reconcileServerService(cr *argoprojv1a1.ArgoCD) error {
 	svc := newServiceWithSuffix("server", "server", cr)
-	if argoutil.IsObjectFound(r.client, cr.Namespace, svc.Name, svc) {
+
+	if err := applyReconcilerHook(cr, svc, ""); err != nil {
+		return err
+	}
+
+	existingSvc := &corev1.Service{}
+	if argoutil.IsObjectFound(r.client, cr.Namespace, svc.Name, existingSvc) {
+		if !reflect.DeepEqual(existingSvc.Annotations, svc.Annotations) {
```

Comment:
If a user adds an annotation to the service, the operator would update the service on each reconcile since `svc.Annotations` would never be equal to `existingSvc.Annotations`. 
Can we perform a check which makes sure that the required annotations are present in the existing service object? and if not present, update it? what do you think?

Reply:
Makes sense. Will add a check

Reply:
Added a check to update only if there are new annotations